spec/active_record_batch_consumer_mysql_spec.rb

```diff
@@ -0,0 +1,219 @@
+# frozen_string_literal: true
+
```

Comment:
The existing test runs with SQLite with `bundle exec rspec`.

The feature of association_list is with MySQL so I had to write this `:integration` test for testing mysql changes.
Will write up a ticket to setup and run integration tests on deimos. 
For Non-MySQL DBs, we raise NotImplementedError which has been tested in the other RSpec.